### PR TITLE
Feature/supabase integration

### PR DESCRIPTION
**Pull request by stephenhungg**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
full supabase integration and scrolling

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 

Screenshot 2025-10-20 at 11 23 17 PM

Screenshot 2025-10-20 at 11 23 44 PM

Screenshot 2025-10-20 at 11 24 07 PM

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

src/supabase folder, src/components, src/screens/PostScreen.tsx, 

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

chatgpt

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @insert pl github username here

### CODE REVIEW

File: src/components/Comment.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  username: string;
+  comment: string;
+  date: string;
+}
+
+interface CommentsForPostProps {
+  postId: string;
+}
+
+export const CommentsForPost: React.FC<CommentsForPostProps> = ({ postId }) => {
```
Comment (kylezryr) [inline src/components/Comment.tsx:18]: this is an outdated version of defining a component in React (using React.FC), you can define it like how you did for `Post.tsx` instead (`export const CommentsForPost = ({ ... }) => { ... })`

**Review by kylezryr — APPROVED** [review]
great work stephen! looks really good on expo go, queries look good and code has no major issues!
